scripts/submit_bb.py

```diff
@@ -36,129 +37,178 @@ def gen_command_template(params):
+    write_directory = write_directory if write_directory else params.sim_dir
+    if write_directory is None:
+        write_directory = params.sim_dir
+
+    underscored_srf = srf_name.replace("/", "__")
+
+    header_dict = {
+        "wallclock_limit": wct,
+        "job_name": f"bb.{underscored_srf}",
+        "job_description": "BB calculation",
+        "additional_lines": "###SBATCH -C avx",
+        "platform_specific_args": get_platform_node_requirements(ncores),
+    }
+
+    body_template_params = (
+        "{}.sl.template".format(sl_name_prefix),
+        {"test_bb_script": "test_bb.sh"},
+    )
 
-        command_template_parameters, add_args = gen_command_template(params)
+    command_template_parameters, add_args = gen_command_template(params)
+
+    script_prefix = f"{sl_name_prefix}_{underscored_srf}"
+    script_file_path = write_sl_script(
+        write_directory,
+        params.sim_dir,
+        const.ProcessType.BB,
+        script_prefix,
+        header_dict,
+        body_template_params,
+        command_template_parameters,
+        add_args,
+    )
 
-        script_prefix = "{}_{}".format(sl_name_prefix, underscored_srf)
-        script_file_path = write_sl_script(
-            write_directory,
+    # Submit the script
+    submit_yes = True if auto else confirm("Also submit the job for you?")
+    if submit_yes:
+        submit_script_to_scheduler(
+            script_file_path,
+            const.ProcessType.BB.value,
+            simulation_structure.get_mgmt_db_queue(params.mgmt_db_location),
             params.sim_dir,
-            const.ProcessType.BB,
-            script_prefix,
-            header_dict,
-            body_template_params,
-            command_template_parameters,
-            add_args,
+            srf_name,
+            target_machine=machine,
+            logger=logger,
         )
 
-        # Submit the script
-        submit_yes = True if args.auto else confirm("Also submit the job for you?")
-        if submit_yes:
-            submit_script_to_scheduler(
-                script_file_path,
-                const.ProcessType.BB.value,
-                simulation_structure.get_mgmt_db_queue(params.mgmt_db_location),
-                params.sim_dir,
-                srf_name,
-                target_machine=args.machine,
-                logger=logger,
-            )
 
+def load_args():
+    """
+    Unpacks arguments and does basic checks
 
-if __name__ == "__main__":
+    Returns
+    -------
+    Processed arguments
+
+    """
     parser = argparse.ArgumentParser(
         description="Create (and submit if specified) the slurm script for BB"
     )
 
-    parser.add_argument("--auto", nargs="?", type=str, const=True)
-    parser.add_argument(
-        "--version",
-        type=str,
-        default=platform_config[const.PLATFORM_CONFIG.BB_DEFAULT_VERSION.name],
-    )
+    parser.add_argument("--auto", action="store_true", default=False)
```

Comment:
This argument should be named --submit or something more descriptive. While it's being changed anyway...

Reply:
This is historical when researchers preferred to interact with the sim configuration.  I'm removing it and renaming --auto  to --submit